gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/ImmunizationMapper.java

```diff
@@ -0,0 +1,301 @@
+                        .asStringValue()));
+            }
+        }
+
+        return null;
+    }
+
+    private Identifier getIdentifier(String id) {
+        return new Identifier()
+            .setSystem(IDENTIFIER_SYSTEM) // TODO: concatenate source practice org id to URL (NIAD-2021)
+            .setValue(id);
+    }
+
+    private boolean hasImmunizationCode(RCMRMT030101UK04ObservationStatement observationStatement) {
+        String snomedCode = observationStatement.getCode().getCodeSystem();
+
+        return IMMUNIZATION_SNOMED_CODE.equals(snomedCode);
+    }
+
+    private Extension createVaccineProcedureExtension(RCMRMT030101UK04ObservationStatement observationStatement) {
+        return new Extension()
+            .setUrl(VACCINE_PROCEDURE_URL)
+            .setValue(new Extension()
+                .setUrl(VACCINATION_CODING_EXTENSION_URL)
+                .setValue(codeableConceptMapper.mapToCodeableConcept(observationStatement.getCode()))
+            );
+    }
+
+    private String getTSStringValue(TS ts) {
+        if (ts == null) {
+            return null;
+        } else if (ts.getValue() != null) {
+            return ts.getValue();
+        } else if (ts.getNullFlavor().equals(CsNullFlavor.UNK)) {
+            return CsNullFlavor.UNK.value();
+        }
+
+        return null;
+    }
+
+    private String getIVXBTSStringValue(IVXBTS ivxbts) {
+        if (ivxbts == null) {
+            return null;
+        } else if (ivxbts.getValue() != null) {
+            return ivxbts.getValue();
+        } else if (ivxbts.getNullFlavor().equals(CsNullFlavor.UNK)) {
+            return CsNullFlavor.UNK.value();
+        }
+
+        return null;
+    }
+
+    private Date getObservationDate(RCMRMT030101UK04ObservationStatement observationStatement) {
+        if (observationStatement.getEffectiveTime() != null) {
+            var center = getTSStringValue(observationStatement.getEffectiveTime().getCenter());
+            var low = getIVXBTSStringValue(observationStatement.getEffectiveTime().getLow());
+            var high = getIVXBTSStringValue(observationStatement.getEffectiveTime().getHigh());
+            var availabilityTimeValue = observationStatement.getAvailabilityTime();
+
+            if (center != null) {
+                return DateFormatUtil.parseToDateTimeType(center).getValue();
+            } else if (low != null && high != null) {
+                return DateFormatUtil.parseToDateTimeType(low).getValue();
+            } else if (low != null) {
+                return DateFormatUtil.parseToDateTimeType(low).getValue();
+            } else if (high != null && availabilityTimeValue.getValue() != null && availabilityTimeValue.getNullFlavor() == null) {
+                return DateFormatUtil.parseToDateTimeType(availabilityTimeValue.getValue()).getValue();
+            }
+        }
+
+        return null;
+    }
+
+    private String highValueToNotes(RCMRMT030101UK04ObservationStatement observationStatement) {
```

Comment:
I think the name of this function is confusing - from the name - I would expect it to add to the notes. Since we're just returning a string value perhaps we should change the name, maybe something like `createDateText`?